### PR TITLE
Upgrade language-docker to version 2.0

### PR DESCRIPTION
**Pull request by lorenzo**
This meant that of the Check rules had to be removed, as they now are parse errors

Fixes https://github.com/hadolint/hadolint/issues/146
Fixes https://github.com/hadolint/hadolint/issues/76
Fixes https://github.com/hadolint/hadolint/issues/53
Fixes https://github.com/hadolint/hadolint/issues/38

### CODE REVIEW

File: src/Hadolint/Rules.hs [hunk 1]
```diff
@@ -311,33 +311,23 @@ useAdd = instructionRule code severity message check
     code = "DL3010"
     severity = InfoC
     message = "Use ADD for extracting archives into an image"
-    check (Copy src dst) = and [not (format `isSuffixOf` src) | format <- archive_formats]
-    check _ = True
-    archive_formats = [".tar", ".gz", ".bz2", "xz"]
-
-exposeMissingArgs = instructionRule code severity message check
```
Comment (lorenzo) [inline src/Hadolint/Rules.hs]: this is now caught as a parse error

File: src/Hadolint/Rules.hs [hunk 2]
```diff
@@ -311,33 +311,23 @@ useAdd = instructionRule code severity message check
...
-  where
-    code = "DL3021"
-    severity = ErrorC
-    message = "EXPOSE requires at least one argument"
-    check (Expose (Ports ports)) = not (null ports)
-    check (Expose (PortStr "")) = False
-    check _ = True
-
-copyMissingArgs = instructionRule code severity message check
```
Comment (lorenzo) [inline src/Hadolint/Rules.hs]: This also became a parse error

File: src/Hadolint/Rules.hs [hunk 3]
```diff
@@ -226,7 +226,7 @@ noLatestTag = instructionRule code severity message check
     message =
         "Using latest is prone to errors if the image will ever update. Pin the version explicitly \
         \to a release tag."
-    check (From (TaggedImage _ tag)) =
+    check (From (TaggedImage _ tag _)) =
```
Comment (zemanlx) [inline src/Hadolint/Rules.hs]: I believe that tag and name are now nicely separated, we can replace this _hack_ by 

```haskell
check (From (TaggedImage _ tag _)) = tag /= "latest"
```

File: src/Hadolint/Rules.hs (hunk 1 above)
Comment (zemanlx) [reply src/Hadolint/Rules.hs]: Can you please also remove it from `README.md`?

File: src/Hadolint/Rules.hs (hunk 2 above)
Comment (zemanlx) [reply src/Hadolint/Rules.hs]: Can you please also remove it from `README.md`?

Comment (lorenzo) [reply src/Hadolint/Rules.hs]: It was not in the readme!

File: src/Hadolint/Rules.hs (hunk 3 above)
Comment (lorenzo) [reply src/Hadolint/Rules.hs]: ah, good catch

**Comment by lukasmartinelli** [comment]
giphy 80

File: src/Hadolint/Rules.hs (hunk 2 above)
Comment (zemanlx) [reply src/Hadolint/Rules.hs]: Sorry, my bad. I was expecting that all checks are in README.

**Comment by lorenzo** [comment]
I'm going to release a new version

**Comment by zemanlx** [comment]
@lorenzo cool, please follow https://github.com/hadolint/hadolint/blob/master/docs/RELEASE.md

**Pull request merged**